_dev/src/ts/types/ComponentAbstract.ts

```diff
@@ -2,7 +2,7 @@
  * @abstract
  * @description Base abstract class defining the structure for page components, requiring implementation of lifecycle methods for mounting and destruction.
  */
-export default abstract class PageAbstract {
+export default abstract class ComponentAbstract {
```

Comment:
I think it's better to leave this abstract has she is and make multiple interface for other use.